pages/blog/subscriptions-and-live-queries-real-time-with-graphql.mdx

```diff
@@ -0,0 +1,1036 @@
+Having to patch a library with `patch-package` before even being usable is generally a bad sign.
+
+1. Automatically updates clients. **Yes.** ✅
+
+When implementing our schema conform to the library, this library delivers precise results once the immutable state has changed.
+
+2. Efficiently updates clients. **Yes.** ✅
+
+Initially a result tree is sent to the client. Afterwards, only JSON patches that must be applied to the initial result are sent to the client.
+
+3. Flexibility. **Kind of.** ✳️
+
+We don't rely on any third party services, however, we forced into immutability to some extend.
+
+4. Adaptability. **It depends.** ✳️
+
+Adding a immutable layer to our existing schema might be a pretty big change. Furthermore, the library does lack support for some GraphQL features such as Interfaces and Unions.
+
+## GraphQL Live Queries via the `@live` directive
+
+There are companies out there, like [Facebook](https://www.youtube.com/watch?v=BSw05rJaCpA), that are already using this approach. There is also a graphql framework available in go that supports live queries out of the box! Check out [thunde](https://github.com/samsarahq/thunder).
+
+The idea behind the `@live` directive is that it is used to mark that the client is interested in keeping that query execution result as up to date as possible. The implementation, however, is up to user-land.
+
+```graphql
+query($id: ID!) @live {
+  post(id: $id) {
+    id
+    title
+    totalLikeCount
+  }
+}
+```
+
+The idea of just making any query without additional overhead a live query is very appealing from the view of a frontend developer.
+From a backend perspective, however that raises new questions.
+Just adding adirective on the operation on frontend won't make the whole backend reactive.
+
+After having built an example app with `graphql-live-subscriptsions` from scratch, studing the flaws of that library and being uncomfortable with the vendor lock-in of services such as Postgraphile and Hasura, I decided to approach the problem of live queries in a more pluggable way, by using the `@live` directive.
+
+### `@n1ru4l/graphql-live-query` A common definition and set of utilities for determining of a live query
+
+This module provides two things.
+
+1. `GraphQLLiveDirective` that can be added to any schema.
+
+```ts
+import { GraphQLSchema, specifiedDirectives } from 'graphql';
+import { GraphQLLiveDirective } from '@n1ru4l/graphql-live-query';
+import { query, mutation, subscription } from './schema';
+
+const schema = new GraphQLSchema({
+  query,
+  mutation,
+  subscription,
+  directives: [
+    GraphQLLiveDirective,
+    /* Keep @defer/@stream/@if/@skip */ ...specifiedDirectives,
+  ],
+});
+```
+
+2. `isLiveQueryOperationDefinitionNode`
+
+This is a simple function that takes a `DocumentNode` and returns `true` if it is a live query document.
```

Comment:
It takes an `OperationDefinitionNode`, not a `DocumentNode` though, right?

Reply:
```suggestion
This is a simple function that takes an `OperationDefinitionNode` and returns `true` if it is a live query document.
```